Block/Embed.php

```diff
@@ -100,4 +100,19 @@ public function getServerAddress()
     {
         return Nosto::getEnvVariable('NOSTO_SERVER_URL', self::DEFAULT_SERVER_ADDRESS);
     }
+
+    /**
+     * Overridden method that only outputs any markup if the extension is enabled and an account
+     * exists for the current store view.
+     *
+     * @return string the markup or an empty string (if an account doesn't exist)
+     */
+    protected function _toHtml()
```

Comment:
I tried it. It's the same issue on M1 and I wasn't able to find a meaningful way to improve this.